### PR TITLE
feat: Add TLS Config options to HTTP template

### PR DESCRIPTION
**Pull request by simster7**
Closes: https://github.com/argoproj/argo-workflows/issues/7390

Signed-off-by: Simon Behar <simbeh7@gmail.com>

Don't bother creating a PR until you've done this:

* [ ] Run `make pre-commit -B` to fix codegen, lint, and commit message problems.

Create your PR as a draft.

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it
  does not need to pass.
* Once required tests have passed, you can make it "Ready for review".
* Say how how you tested your changes. If you changed the UI, attach screenshots.

Tips:

* If changes were requested, and you've made them, then dismiss the review to get it looked at again.
* Add you organization to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md) if you like.
* You can ask for help!

### CODE REVIEW

File: pkg/apis/workflow/v1alpha1/http_types.go [hunk 1]
```diff
@@ -43,4 +43,13 @@ type HTTP struct {
...
 	Body string `json:"body,omitempty" protobuf:"bytes,5,opt,name=body"`
+	// TLSConfig are configuration options for TLS
+	TLSConfig *TLSConfig `json:"tlsConfig,omitempty"`
+}
+
+type TLSConfig struct {
+	// Insecure is a flag to allow insecure connections
+	Insecure bool `json:"insecure,omitempty"`
+	// CACertificateKeyRef is a secret key reference to RootCA certificates
+	CACertificateKeyRef *v1.SecretKeySelector `json:"caCertificateKeyRef,omitempty"`
```
Comment (simster7) [inline pkg/apis/workflow/v1alpha1/http_types.go:54]: Please review config spec

**Review by simster7 — COMMENTED** [review]
Needs tests

File: pkg/apis/workflow/v1alpha1/http_types.go (hunk 1 above)
Comment (alexec) [reply pkg/apis/workflow/v1alpha1/http_types.go:54]: should be consistent with `valueFrom`?

**Comment by agilgur5** [comment]
Replaced by #7929

**Pull request closed without merging**